From 14c40e5bb17db311792b8457a9a2e1e8bd15b3f4 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Sun, 5 Mar 2023 20:59:19 +0100
Subject: [PATCH] ARM: dts: imx: Add WDT bindings on DH i.MX6 DHSOM

Add WDT reboot bindings on DH i.MX6 DHSOM to permit the platform
to reboot via WDT in U-Boot. These are custom U-Boot bindings,
hence they are placed in -u-boot.dtsi .

Upstream-Status: Submitted [https://patchwork.ozlabs.org/project/uboot/patch/20230305204908.772661-1-marex@denx.de/]
Signed-off-by: Marek Vasut <marex@denx.de>
---
Cc: Andreas Geisreiter <ageisreiter@dh-electronics.de>
Cc: Christoph Niedermaier <cniedermaier@dh-electronics.com>
Cc: Fabio Estevam <festevam@denx.de>
Cc: Peng Fan <peng.fan@nxp.com>
Cc: Stefano Babic <sbabic@denx.de>
Cc: u-boot@dh-electronics.com
---
 arch/arm/dts/imx6qdl-dhcom-u-boot.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/dts/imx6qdl-dhcom-u-boot.dtsi b/arch/arm/dts/imx6qdl-dhcom-u-boot.dtsi
index 190567ab7ba..06ad98bccf6 100644
--- a/arch/arm/dts/imx6qdl-dhcom-u-boot.dtsi
+++ b/arch/arm/dts/imx6qdl-dhcom-u-boot.dtsi
@@ -8,6 +8,12 @@
 	aliases {
 		eeprom0 = &eeprom0;
 	};
+
+	wdt-reboot {
+		compatible = "wdt-reboot";
+		wdt = <&wdog1>;
+		u-boot,dm-spl;
+	};
 };
 
 &fec {
@@ -25,3 +31,7 @@
 	gpio = <&gpio3 31 GPIO_ACTIVE_HIGH>;
 	enable-active-high;
 };
+
+&wdog1 {
+	u-boot,dm-spl;
+};
-- 
2.39.2

